Tests/DatadogTests/Datadog/Tracing/ContextMessageReceiverTests.swift

```diff
@@ -0,0 +1,63 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import XCTest
+import TestUtilities
+
+@testable import Datadog
+
+class ContextMessageReceiverTests: XCTestCase {
+    func testItReceivesApplicationStateHistory() throws {
+        // Given
+        let receiver = ContextMessageReceiver(bundleWithRUM: .mockRandom())
+        let core = PassthroughCoreMock(
+            context: .mockWith(applicationStateHistory: .mockAppInBackground()),
+            messageReceiver: receiver
+        )
+
+        XCTAssertEqual(receiver.context.applicationStateHistory?.initialSnapshot.state, .background)
+
+        // When
+        core.context.applicationStateHistory.append(.init(state: .active, date: Date()))
+
+        // Then
+        XCTAssertEqual(receiver.context.applicationStateHistory?.currentSnapshot.state, .active)
+    }
+
+    func testItReceivesRUMContext() throws {
+        // Given
+        let receiver = ContextMessageReceiver(bundleWithRUM: true)
+        let core = PassthroughCoreMock(
+            context: .mockWith(featuresAttributes: ["rum": ["ids": ["key": "value1"]]]),
+            messageReceiver: receiver
+        )
+
+        XCTAssertEqual(receiver.context.rum, ["key": "value1"])
+
+        // When
+        core.set(feature: "rum", attributes: { ["ids": ["key": "value2"]] })
+
+        // Then
+        XCTAssertEqual(receiver.context.rum, ["key": "value2"])
+    }
```

Comment:
question/ Shouldn't we define such tests as Integration Units? Here it makes very little sense - the contract between RUM and Tracing that we mock here is arbitrary and false (RUM context is more than `rum.ids[key] = value1`). We don't test much more than basic V2's `MessageReceiver` logic here.

Reply:
This test is more to check the `ContextMessageReceiver(bundleWithRUM: true)` vs. `ContextMessageReceiver(bundleWithRUM: false)`, and assert that `bundleWithRUM` enables/disables the rum context propagation. So we are testing the receiver only in unit test.

Now, I agree that we should be testing the integration RUM+Trace. It was previously covered, but we have a regression that will be addressed in [RUMM-2843](https://datadoghq.atlassian.net/browse/RUMM-2843). We will re-enable that coverage while fixing it. How does that sound?